src/Compilers/Core/Portable/SourceGeneration/GeneratorDriver.cs

```diff
@@ -269,7 +271,7 @@ internal GeneratorDriverState RunGeneratorsCore(Compilation compilation, Diagnos
                 }
             }
 
-            state = state.With(stateTable: driverStateBuilder.ToImmutable(), generatorStates: stateBuilder.ToImmutableAndFree(), runTime: timer.Elapsed);
+            state = state.With(stateTable: driverStateBuilder.ToImmutable(), syntaxStore: syntaxStoreBuilder.ToImmutable(), generatorStates: stateBuilder.ToImmutableAndFree(), runTime: timer.Elapsed);
```

Comment:
We have two distinct instances of `SyntaxStore` - from `driverStateBuilder.ToImmutable().SyntaxStore` and from `syntaxStoreBuilder.ToImmutable()`. If it's necessary to pass a `SyntaxStore` instance directly to `With()`, consider refactoring to:
```suggestion
            var stateTable = driverStateBuilder.ToImmutable();
            state = state.With(stateTable, stateTable.SyntaxStore, generatorStates: stateBuilder.ToImmutableAndFree(), runTime: timer.Elapsed);
```

Reply:
`DriverStateTable` doesnt actually hold onto the `SyntaxStore` so we only have the single instance.